base/linalg/eigen.jl

```diff
@@ -90,11 +90,17 @@ end
 #Computes maximum and minimum eigenvalue
 function eigmax(A::Union{Number, StridedMatrix}; permute::Bool=true, scale::Bool=true)
     v = eigvals(A, permute = permute, scale = scale)
-    iseltype(v,Complex) ? error("DomainError: complex eigenvalues cannot be ordered") : maximum(v)
+    if iseltype(v,Complex)
+      throw(DomainError())
```

Comment:
Oh, bummer, `DomainError` doesn't have an extra message field like I thought it would. (So https://github.com/JuliaLang/julia/blob/71f08f30c915d3087c966daf66dd17c01b9092c5/base/linalg/cholesky.jl#L89 is probably untested.) Throwing a more specific exception type is good, but so is providing additional info to the user. Looks like `DomainError`s get specially handled here https://github.com/JuliaLang/julia/blob/71f08f30c915d3087c966daf66dd17c01b9092c5/base/replutil.jl#L96-L113